terraform/aws/modules/infra/network.tf

```diff
@@ -21,9 +21,9 @@ resource "aws_security_group" "batch" {
 
   egress {
     description = "Allow ECS tasks to talk to the internet"
-    from_port   = 443
-    to_port     = 443
-    protocol    = "tcp"
+    from_port   = 0
+    to_port     = 0
+    protocol    = "-1"
```

Comment:
So here is why this branch took so long to work out: the security group rule for ECS tasks had been configured to allow egress over HTTPS only (that's port 443). At some point this summer, the `snowflake-connector-python` package was updated, and in those updates it started making a few exploratory connections over HTTP (port 80). I think these connections are just sniffing out Snowflake connectivity and features, but to be honest it's all under the hood, so I'm not really sure.

So in order for these ECS tasks to start talking to Snowflake again, we need to open up the security group to talk over more ports. Here I'm just allowing outgoing connections to the internet over any port (incoming requests are still restricted).